### ISSUE TITLE
Namevars detected as optional parameters

### ISSUE THREAD

**Issue by ntnn** [rodjek/puppet-lint#633]
Example:
```puppet
define example (
    String $somevar = $name,
    String $anothervar,
) {
    notify { $somevar: }
}

example { 'name':
  anothervar => 'something',
}
```
command:
```sh
puppet-lint --no-autoloader_layout-check --no-strict_indent-check --no-documentation-check example.pp
```

Expected output: none
Actual output:
```
WARNING: optional parameter listed before required parameter on line 3
```

I'd argue that a namevar for defined types are still required parameters, since the title of a resource is required, hence the parameter is still required, even though it has a default, which is in turn a required parameter.

**Comment by binford2k** [comment]
Until this is allowed by the style guide, we cannot make this change. There is a discussion ticket open https://github.com/puppetlabs/puppet-docs/pull/725

**Comment by rodjek** [comment]
The unanimous decision of the discussion in puppetlabs/puppet-docs#725 was that although you're defaulting your parameter to a required value, the parameter itself is still optional as it's not necessary for someone to specify a value for it.

Given that, the check output is correct in this case and I'm going to close this issue.
